### PR TITLE
Use std::mutex, not boost::mutex

### PR DESCRIPTION
**Pull request by Al2Klimov**

### CODE REVIEW

File: lib/base/timer.cpp [hunk 1]
```diff
@@ -295,11 +299,11 @@ void Timer::TimerThreadProc()
 		auto it = idx.begin();
 		Timer *timer = *it;
 
-		double wait = timer->m_Next - Utility::GetTime();
+		auto next (ch::system_clock::from_time_t(0) + ch::duration<double>(timer->m_Next));
```
Comment (julianbrost) [inline lib/base/timer.cpp:302]: Why not use the constructor of `time_point` that takes a duration after the clock's epoch? See (2) on https://en.cppreference.com/w/cpp/chrono/time_point/time_point

**Review by julianbrost — CHANGES_REQUESTED** [review]
If `boost::mutex` and `boost::condition_variable` are gone, it should no longer be necessary to include the corresponding `boost/thread/mutex.hpp` and `boost/thread/condition_variable.hpp` headers. Can you please also remove these includes?

File: lib/base/singleton.hpp [hunk 2]
```diff
@@ -4,7 +4,7 @@
 #define SINGLETON_H
 
 #include "base/i2-base.hpp"
-#include <boost/thread/mutex.hpp>
+#include <mutex>
```
Comment (julianbrost) [inline lib/base/singleton.hpp:7]: This include shouldn't be needed here at all.

File: lib/livestatus/logtable.hpp [hunk 3]
```diff
@@ -4,7 +4,7 @@
 #define LOGTABLE_H
 
 #include "livestatus/historytable.hpp"
-#include <boost/thread/mutex.hpp>
+#include <mutex>
```
Comment (julianbrost) [inline lib/livestatus/logtable.hpp:7]: Also doesn't seem to be required at all for this file.

File: lib/livestatus/statehisttable.hpp [hunk 4]
```diff
@@ -5,7 +5,7 @@
 
 #include "icinga/service.hpp"
 #include "livestatus/historytable.hpp"
-#include <boost/thread/mutex.hpp>
+#include <mutex>
```
Comment (julianbrost) [inline lib/livestatus/statehisttable.hpp:8]: Seems to not be required for this file at all.

File: lib/base/logger.hpp [hunk 5]
```diff
@@ -6,7 +6,7 @@
 #include "base/i2-base.hpp"
 #include "base/logger-ti.hpp"
 #include <set>
-#include <iosfwd>
+#include <sstream>
```
Comment (julianbrost) [inline lib/base/logger.hpp:9]: I don't get this this change. There's no change regarding any iostreams, so why is this needed?

**Review by julianbrost — CHANGES_REQUESTED** [review]

File: lib/base/logger.hpp (hunk 5 above)
Comment (Al2Klimov) [reply lib/base/logger.hpp:9]: https://github.com/Icinga/icinga2/runs/1813342849?check_suite_focus=true

Comment (julianbrost) [reply lib/base/logger.hpp:9]: Ok, and this was probably included by some of the two boost headers which was basically included everywhere. Fine.

**Review by julianbrost — APPROVED** [review]
